lib/text/vtt_text_parser.js

```diff
@@ -219,6 +221,46 @@ shaka.text.VttTextParser = class {
     return cue;
   }
 
+  /**
+   * Parses a WebVTT styles from the given payload.
+   *
+   * @param {!shaka.text.Cue} cue
+   * @param {string} payload
+   */
+  static parseCueStyles(cue, payload) {
+    try {
+      const parser = new DOMParser();
+      const xml = parser.parseFromString(payload, 'text/xml');
+      const parserError = xml.getElementsByTagName('parsererror')[0];
+      if (parserError) {
```

Comment:
You could use `shaka.util.XmlUtils.parseXmlString` for this.  You'll probably need to edit it to allow arbitrary root elements.

Reply:
OK, I'll change it